src/test/java/com/KacFlor/ShopSpring/service/CustomerServiceTest.java

```diff
@@ -0,0 +1,188 @@
+package com.KacFlor.ShopSpring.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.KacFlor.ShopSpring.controllersRequests.CustomerUpdateRequest;
+import com.KacFlor.ShopSpring.dao.CustomerRepository;
+import com.KacFlor.ShopSpring.dao.UserRepository;
+import com.KacFlor.ShopSpring.model.Customer;
+import com.KacFlor.ShopSpring.model.Role;
+import com.KacFlor.ShopSpring.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+import org.springframework.security.core.Authentication;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomerServiceTest{
+
+    @Mock
+    private CustomerRepository customerRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private CustomerService customerService;
+
+    private User user1;
+
+    private User user2;
+
+    private User hAdmin;
+
+    private Customer customer = new Customer();
+
+    private Customer customer1 = new Customer();
+
+    private Customer customer2 = new Customer();
+
+    @BeforeEach
+    public void setup(){
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
+        user1 = new User("Test1", "encodedPassword1", customer1, Role.USER);
+        user2 = new User("Test2", "encodedPassword2", customer2, Role.USER);
+        hAdmin = new User("HAdmin", encoder.encode("1234"), customer, Role.ADMIN);
+        customer1.setUser(user1);
+        customer2.setUser(user2);
+        customer.setUser(hAdmin);
+        customer1.setFirstName("Test1");
+        customer2.setFirstName("Test2");
+        customer.setFirstName("HAdmin");
```

Comment:
Why are these variables needed, where are they used?

Reply:
Yes in my logic i made that after you register user program create new customer to this user and set firstname equal to login from user. I use it inside testDeleteCustomer line 178 when(customerRepository.findByFirstName("Test1")).thenReturn(customer1); for example.

Reply:
look my comment above

Reply:
done